File: Done/Binary_Search_307_Questions/98/Solution.java
class Solution {
    public int closestToTarget(int[] arr, int target) {
        int ret = Integer.MAX_VALUE;
        for (int i = 0; i < arr.length; i++){
            ret = Math.min(ret, Math.abs(arr[i] - target));
            for (int j = i-1; j >= 0 && (arr[j] & arr[i]) != arr[j]; j--){
                arr[j] &= arr[i];
                ret = Math.min(ret, Math.abs(arr[j] - target));
            } 
        }
        return ret;
        
    }
}

File: Done/Binary_Search_307_Questions/98/Solution.py
class Solution:
    def closestToTarget(self, arr: List[int], target: int) -> int:
        ans, seen = inf, set()
        for x in arr: 
            seen = {ss & x for ss in seen} | {x}
            ans = min(ans, min(abs(ss - target) for ss in seen))
        return ans 